CONTRIBUTING.md

```diff
@@ -0,0 +1,219 @@
+# Contribute to the secrets of sonar-text
+
+## Development environment
+
+1. Set up a Java development environment, or use Docker, to be ready to perform commands such as those specified in the [README.md](README.md)
+2. Set up an extension that validates YAML files, such as [YAML by RedHat](https://github.com/redhat-developer/vscode-yaml)
+3. Set up this [json-schema file](https://github.com/SonarSource/sonar-text/blob/master/sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/specifications/specification-json-schema.json) for your validation tooling
+4. Take a look at [the existing rules](https://github.com/SonarSource/sonar-text/tree/master/sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration)
+5. Familiarize yourself with [SonarSource/rspec](https://github.com/SonarSource/rspec)
+
+## Adding a secret
+
+1. Write the yaml file
+2. Validates the yaml against the schema file
+3. use the script `secretSpecificationInclusionGenerator.sh <Rule-Api jar>`
+  1. It will generate the java files serving as a "glue" between the yaml file and the code base
+  2. If the rspec text code is already available in the master branch of [SonarSource/rspec](https://github.com/SonarSource/rspec), it will also generate additional rspec data
+4. Test the code:
+  1. If rspec data is available, do `mvn clean test`
+  2. Else, do `mvn test -Dtest=FakecloudCheckTest` and replace this test name by the one generated previously 
+5. When rspec is finally merged:
+   ``` bash
+   pushd sonarpedia-secrets
+   java -jar ../rule-api-2.7.0.2609.jar generate -rule SXXXX
+   popd
+   ```
```

Comment:
The commands here have now been replaced with different ones that are also included in the README (except the single test).